### PR TITLE
Add localized files & packaging support

### PR DESCRIPTION
**Pull request by j0shuams**
This PR adds the files TDBuild localized, puts them through the CodeSigning task on the pipeline, and provides packaging support to include the files in our NuGet.

### CODE REVIEW

File: nuget/Microsoft.Windows.CsWinRT.nuspec [hunk 1]
```diff
@@ -36,12 +36,22 @@
...
     <file src="$winrt_host_arm64$" target ="runtimes\win-arm64\native"/>
-    <file src="$winrt_host_resource_x64$" target ="runtimes\win-x64\native\en-us"/>
-    <file src="$winrt_host_resource_x86$" target ="runtimes\win-x86\native\en-us"/>
-    <file src="$winrt_host_resource_arm$" target ="runtimes\win-arm\native\en-us"/>
-    <file src="$winrt_host_resource_arm64$" target ="runtimes\win-arm64\native\en-us"/>
     <file src="$winrt_shim$" target ="lib\net5.0\"/>
+
+    <!-- Localized resources for error strings --> 
     
+    <file src="..\src\Authoring\WinRT.Host\MUI\**" target ="runtimes\win-x64\native"/>
```
Comment (manodasanW) [inline nuget/Microsoft.Windows.CsWinRT.nuspec:43]: Is this just the MUI file or others too?

File: nuget/Microsoft.Windows.CsWinRT.nuspec [hunk 2]
```diff
@@ -36,12 +36,22 @@
...
+    <file src="..\src\Authoring\WinRT.Host\MUI\**" target ="runtimes\win-x86\native"/>
+    <file src="..\src\Authoring\WinRT.Host\MUI\**" target ="runtimes\win-arm\native"/>
+    <file src="..\src\Authoring\WinRT.Host\MUI\**" target ="runtimes\win-arm64\native"/>
+    <file src="$winrt_host_resource_x64$" target ="runtimes\win-x64\native\en-US"/>
```
Comment (manodasanW) [inline nuget/Microsoft.Windows.CsWinRT.nuspec:47]: What is the difference between this and the above ones again?

File: nuget/Microsoft.Windows.CsWinRT.nuspec (hunk 1 above)
Comment (manodasanW) [reply nuget/Microsoft.Windows.CsWinRT.nuspec:43]: I see below, it is just the MUI with folder path included/

File: nuget/Microsoft.Windows.CsWinRT.nuspec (hunk 2 above)
Comment (manodasanW) [reply nuget/Microsoft.Windows.CsWinRT.nuspec:47]: Is this an optimization for en-US?

File: src/Authoring/WinRT.SourceGenerator/ResX/fr-FR/CsWinRTDiagnosticStrings.resx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
```
Comment (manodasanW) [inline src/Authoring/WinRT.SourceGenerator/ResX/fr-FR/CsWinRTDiagnosticStrings.resx:48]: I expect there will be a future PR that populates it with localized content?

Comment (j0shuams) [reply src/Authoring/WinRT.SourceGenerator/ResX/fr-FR/CsWinRTDiagnosticStrings.resx:48]: We got the rest of them now, including them in this PR as well.

File: nuget/Microsoft.Windows.CsWinRT.nuspec (hunk 2 above)
Comment (j0shuams) [reply nuget/Microsoft.Windows.CsWinRT.nuspec:47]: The MUI folder in the repo just has folders like "fr-FR". So we're matching that pattern by putting the default ones (built when the project builds) in the english folder next to the french one.

File: build/AzurePipelineTemplates/CsWinRT-PublishToNuget-Stage.yml [hunk 4]
```diff
@@ -94,6 +94,9 @@ stages:
           release_arm64\WinRT.Host.dll
           net5.0\IIDOptimizer\IIDOptimizer.exe
           net5.0\IIDOptimizer\IIDOptimizer.dll
+          Authoring\WinRT.Host\MUI\**\WinRT.Host.dll.mui
```
[the remark this answers is not in the document]
Comment (j0shuams) [reply build/AzurePipelineTemplates/CsWinRT-PublishToNuget-Stage.yml:97]: I'd rather do all the signing in one place, is there a reason you think it'd be better to do it separately?

[the remark this answers is not in the document]
Comment (j0shuams) [reply build/AzurePipelineTemplates/CsWinRT-PublishToNuget-Stage.yml]: Okay sounds good, I've made the change and spun another build. Will replace the files on this branch with the signed ones that come off the pipeline.

File: build/AzurePipelineTemplates/CsWinRT-PublishToNuget-Stage.yml [hunk 5]
```diff
@@ -90,7 +90,9 @@ stages:
           net5.0\WinRT.Runtime.dll
           net5.0\WinRT.Host.Shim.dll
           release_x64\WinRT.Host.dll
+          release_x64\WinRT.Host.dll.mui
           release_x86\WinRT.Host.dll
+          release_x86\WinRT.Host.dll.mui
           release_arm64\WinRT.Host.dll
```
Comment (manodasanW) [inline build/AzurePipelineTemplates/CsWinRT-PublishToNuget-Stage.yml:96]: any needed for arm64?

File: nuget/Microsoft.Windows.CsWinRT.nuspec [hunk 6]
```diff
@@ -36,12 +36,24 @@
...
-    <file src="$winrt_host_resource_arm$" target ="hosting\arm\native\en-us"/>
-    <file src="$winrt_host_resource_arm64$" target ="hosting\arm64\native\en-us"/>
     <file src="$winrt_shim$" target ="lib\net5.0\"/>
-    
+
+    <!-- Localized resources for error strings 
+         The paths here leads to a folder of folders, 
+           wherein the folders are the resources for a particular locale/cultures. 
+         E.g. fr-FR/WinRT.Host.dll.mui -->
+    <file src="..\src\Authoring\WinRT.Host\MUI\**" target ="hosting\x64\native"/>
```
Comment (manodasanW) [inline nuget/Microsoft.Windows.CsWinRT.nuspec:45]: We should have a follow up PR to demonstrate this in use.

**Review by manodasanW — APPROVED** [review]

**Pull request merged**